src/test/ui/consts/const_discriminant.rs

```diff
@@ -0,0 +1,36 @@
+// run-pass
+#![feature(const_discriminant)]
+#![allow(dead_code)]
+
+use std::mem::{discriminant, Discriminant};
+
+fn identity<T>(x: T) -> T { x }
```

Comment:
Please also add `#[inline(never)]` to this function to ensure the inliner will never optimize this out